apis/dns/v1alpha1/dns_types.go

```diff
@@ -0,0 +1,397 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// ZoneParameters define the desired state of an Azure DNS Zone.
+type ZoneParameters struct {
+	// ResourceGroupName specifies the name of the resource group that should
+	// contain this DNS Zone.
+	// +immutable
+	ResourceGroupName string `json:"resourceGroupName,omitempty"`
+
+	// ResourceGroupNameRef - A reference to a ResourceGroup object to retrieve
+	// its name
+	// +immutable
+	ResourceGroupNameRef *xpv1.Reference `json:"resourceGroupNameRef,omitempty"`
+
+	// ResourceGroupNameSelector - A selector for a ResourceGroup object to
+	// retrieve its name
+	// +immutable
+	ResourceGroupNameSelector *xpv1.Selector `json:"resourceGroupNameSelector,omitempty"`
+
+	// Location is the Azure location that the DNS Zone will be created in
+	// +required
```

Comment:
```suggestion
	// kubebuilder:validation:Required
```
Also applies to similar others.